packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -69,34 +71,64 @@ import {
 
 const compareSrs = createCompare([isSrShared])
 
+@connectStore(() => ({
+  checkPermissions: getCheckPermissions,
+  isAdmin,
+}))
 class VdiSr extends Component {
+  _getIsSrAdmin = createSelector(
+    () => this.props.checkPermissions,
+    () => getDefined(() => this.props.vdiSr.id),
+    (check, sr) => check(sr, 'administrate')
+  )
+
   _getCompareContainers = createSelector(
+    () => this.props.isAdmin,
     () => this.props.userData.vm.$pool,
-    poolId => createCompareContainers(poolId)
+    (isAdmin, poolId) => (isAdmin ? createCompareContainers(poolId) : undefined)
   )
 
   _getSrPredicate = createSelector(
     () => this.props.userData.vm.$pool,
     poolId => sr => sr.$pool === poolId && isSrWritable(sr)
   )
 
-  _onChangeSr = sr => migrateVdi(this.props.item.vdi, sr)
+  _onChangeSr = sr => {
+    const {
+      item: { vdi },
+      userData: { resourceSet },
+    } = this.props
+    return migrateVdi(
+      vdi,
+      sr,
+      getDefined(() => resourceSet.id)
+    )
+  }
 
   render() {
-    const sr = this.props.item.vdiSr
+    const {
+      item: { vdiSr },
+      userData: { resourceSet },
+    } = this.props
+    const isSrAdmin = this._getIsSrAdmin()
```

Comment:
Why not `isSelf` like you did here  https://github.com/vatesfr/xen-orchestra/blob/4322409cc3170ebe1065976f82020c185f540512/packages/xo-web/src/xo-app/vm/tab-disks.js#L512

Reply:
I need to check the SR permission here to display the SR for admin and also for the user with ACLs.

Reply:
We don't need to check permissions for this. 

If the user has a permission on the SR, the SR will appear if not it will not appear. This  is handled by `render-xo-item/SR'. 

What we need to handle, is the self users.

Reply:
I need it  for the `link` and `self` props  (the way that it will be displayed) and for the `xoType`.